### DISCUSSION TITLE
Is there a way to install all requirements for all custom nodes at the same time?

### DISCUSSION THREAD

**Discussion by Vektor369** [Comfy-Org/ComfyUI#6019] [Q&A] [answered]
Re-installing comfy, and I have a lot of custom nodes. Going through each one to install the requirements is a bit tedious.

Thanks in advance.

**Comment by wibur0620** [comment]
No, because a lot of plug-ins, they rely on versions, which would create conflicts.

**Comment by Vektor369** [comment]
So basically, you have to re-install all of the custom nodes individually, one at a time.

**Reply by wibur0620** [reply, marked as answer]
Yes, and you'll also need to check whether the plugin's dependencies are in conflict with your existing dependencies. (Most people don't check, but getting this habit will help you understand errors caused by a dependency conflict)

**Reply by Vektor369** [reply]
That is insane. With the frequency that it breaks at any given moment, to have to reinstall every single custom node, one at a time... That's ridiculous. I have over 70 nodes, and I'm sure that's a very low number compared to most! There's got to be a better way, that's all I'm saying.

**Reply by wibur0620** [reply]
I divide the plug-ins into two categories. One is a tool plug-in (which handles simple tasks and doesn't require model involvement). This type of plug-in basically doesn 't rely on libraries to install. This type of plugin just dragged into the plug-in folder. Install more troublesome is another type of plug-in (this kind of plug-in is generally for the model service, more new open source model in order to use in comfyui, Some authors write comfyui plugins because companies that model open source have certain dependencies and specific version numbers when they open code. The plugin author uses the same dependency library for the plugin to work well.) This type of plugin will install the dependency library when installed. If this dependency is already in your comfyui, And it does not conflict with the version of the dependencies required for the plug-in, so the plug-ins will run fine without installing the dependencies again. If the dependencies conflict, then the plug-int will conflict with some other plug-ins. At this point, it's up to you to make your trade-offs with these plug-ins, or cancel the version number of the dependency library when installing the plug-ins to see if the startup reports an error. Like you said, you have 70 plug-ins. You can first drag the plug-ins that don't rely on libraries into the plug-in folder, Then start to see if it's working properly, and then install the plug-ins that have dependencies one by one. It's best to do this. If you install them all at once, there's a chance you'll report an error because of the conflict. Errors due to dependencies are inevitable in comfyui because all open source models have different dependencies.

**Reply by Vektor369** [reply]
That's sound advice. At this point, I'm honestly considering creating a separate instance of Comfy for each workflow that I use, then keep one 'clean' version for anything new. If something new works out, and I want to keep it, create a new instance, and so forth. I know that sounds excessive, and like a pain, but I'm just so tired of it breaking all the time. Everything will be working fine one day, then I open it the next, and everything's broken again. I've honestly spent so much time chasing down errors, trying to fix it, that it gets to feeling like I spend more time troubleshooting it rather than using it. And I am **not** someone who knows what they're doing, so it just seems like separate instances might be worth my while.

**Comment by ltdrdata** [comment]
FYI, you can utilize `restore-dependencies` command.
https://github.com/ltdrdata/ComfyUI-Manager/blob/main/docs/en/cm-cli.md

**Reply by kiriri** [reply]
This should be marked as the answer. It does exactly what the OP wanted.  
To explain a bit more :  
  
- Open your new comfy-ui folder in the terminal.
- Make sure you have activated the correct virtual environment.
- Run `python ./custom_nodes/ComfyUI-Manager/cm-cli.py restore-dependencies` .
  
That's it, all your dependencies will be installed automatically.
